homeassistant/components/media_player/emby.py

```diff
@@ -0,0 +1,370 @@
+        return 'emby_{}'.format(self.device['DeviceName']) or \
+            DEVICE_DEFAULT_NAME
+
+    @property
+    def session(self):
+        """Return the session, if any."""
+        if self.device['DeviceId'] not in self.emby_sessions:
+            return None
+
+        return self.emby_sessions[self.device['DeviceId']]
+
+    @property
+    def now_playing_item(self):
+        """Return the currently playing item, if any."""
+        session = self.session
+        if session is not None and 'NowPlayingItem' in session:
+            return session['NowPlayingItem']
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        session = self.session
+        if session:
+            if 'NowPlayingItem' in session:
+                if session['PlayState']['IsPaused']:
+                    return STATE_PAUSED
+                else:
+                    return STATE_PLAYING
+            else:
+                return STATE_IDLE
+        # This is nasty. Need to find a way to determine alive
+        else:
+            return STATE_OFF
+
+        return STATE_UNKNOWN
+
+    def update(self):
+        """Get the latest details."""
+        self.update_devices(no_throttle=True)
+        self.update_sessions(no_throttle=True)
+
+    def play_percent(self):
+        """Return current media percent complete."""
+        if self.now_playing_item['RunTimeTicks'] and \
+                self.session['PlayState']['PositionTicks']:
+            return int(self.session['PlayState']['PositionTicks']) / \
+                int(self.now_playing_item['RunTimeTicks']) * 100
+        else:
+            return 0
+
+    @property
+    def app_name(self):
+        """Return current user as app_name."""
+        # Ideally the media_player object would have a user property.
+        try:
+            return self.device['UserName']
+        except KeyError:
+            return None
+
+    @property
+    def media_content_id(self):
+        """Content ID of current playing media."""
+        if self.now_playing_item is not None:
+            return self.now_playing_item['Id']
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        if self.now_playing_item is None:
+            return None
+        media_type = self.now_playing_item['Type']
+        if media_type == 'Episode':
+            return MEDIA_TYPE_TVSHOW
+        elif media_type == 'Movie':
+            return MEDIA_TYPE_VIDEO
+        return None
+
+    @property
+    def media_duration(self):
+        """Duration of current playing media in seconds."""
+        if self.now_playing_item and self.media_content_type:
+            return int(self.now_playing_item['RunTimeTicks']) / 10000000
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        if self.now_playing_item is None:
+            return None
+        media_type = self.now_playing_item['Type']
```

Comment:
Just starting testing this and it certainly does work well.  I'll make the change.